lib/ex_phone_number/validation.ex

```diff
@@ -112,13 +112,49 @@ defmodule ExPhoneNumber.Validation do
     end
   end
 
-  def test_number_length_against_pattern(pattern, number) do
-    if matches_entirely?(pattern, number) do
-      ValidationResults.is_possible
+  def test_number_length(number, metadata) do
+    test_number_length_for_type(number, metadata, PhoneNumberTypes.unknown)
+  end
+
+  def test_number_length_for_type(number, metadata, type) do
+    desc_for_type = get_number_description_by_type(metadata, type)
+    desc_general = get_number_description_by_type(metadata, :general)
+
+    possible_lengths = case Enum.count(desc_for_type.possible_lengths) do
+      0 -> desc_general.possible_lengths
+      _ -> desc_for_type.possible_lengths
+    end
+
+    possible_lengths =
```

Comment:
I would still add 

        |> Enum.sort
        |> Enum.dedup

to the part using `++` operator. Currenlty tests are passing without it, but I think it's more robust code, to avoid having multiple values in returned lists.

Reply:
Sure, you can also just use [Enum.uniq/1](https://hexdocs.pm/elixir/Enum.html#uniq/1). No need to sort and dedup.

Reply:
Well, since after that I use 

    min_length = List.first(possible_lengths)
    max_length = List.last(possible_lengths)

I think it would be a problem, potentially, not to sort the list.

Reply:
Then maybe go for `Enum.min` and `Enum.max`. Possible lengths are integer values, so then it would be rather easy to understand what we want from those. With sorting a few lines back, the reason for it is not clear at that point and min/max from enum do the same and intent is clear. I think 😄